warehouse/query-core/src/main/java/datawave/query/tld/TLDQueryIterator.java

```diff
@@ -171,17 +171,11 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
             if (!startKey.equals(range.getStartKey())) {
                 Key endKey = range.getEndKey();
                 boolean endKeyInclusive = range.isEndKeyInclusive();
-                // if the start key is outside of the range, then reset the end key to the next key
-                if (range.afterEndKey(startKey)) {
-                    endKey = startKey.followingKey(PartialKey.ROW_COLFAM);
-                    endKeyInclusive = false;
-                }
                 range = new Range(startKey, false, endKey, endKeyInclusive);
```

Comment:
wont this restart the current document?

Reply:
nope.  The TLD.getNextParentKey ensures that.  I am just removing the nasty logic that would advance the end key resulting in a range mutually exclusive with the original range.  That is where we got into trouble because the "FinalDocument" key would return a key using the start key of this range which would result in the iterator returning a key outside the original range.  I would rather fail here quickly if we attempt to create a start key past the original end key.  Setting the "isTld" flag correctly results in the TupleToRange class creating a range that is large enough that this will not happen.